geode-core/src/test/java/org/apache/geode/cache/query/internal/ArithmeticOperationsIntegrationTest.java

```diff
@@ -0,0 +1,240 @@
+      fail();
+    } catch (QueryInvocationTargetException qite) {
+      assertEquals("Arithmetic Operations can only be applied to numbers", qite.getMessage());
+    }
+  }
+
+  @Test
+  public void subtractOperationShouldThrowExceptionWhenAddingNull() throws Exception {
+    try {
+      SelectResults results = (SelectResults) queryService
+          .newQuery("select * from /portfolio p where p.ID - null = 2").execute();
+      fail();
+    } catch (QueryInvocationTargetException qite) {
+      assertEquals("Cannot evaluate arithmetic operations on null values", qite.getMessage());
+    }
+  }
+
+  @Test
+  public void subtractOperationShouldCorrectlySubtractInQuery() throws Exception {
+    SelectResults results = (SelectResults) queryService
+        .newQuery("select * from /portfolio p where (p.ID - 1) = 2").execute();
+
+    assertEquals(1, results.size());
+  }
+
+  @Test
+  public void multiplyOperationShouldThrowExceptionWhenAddingNonNumeric() throws Exception {
+    try {
+      SelectResults results = (SelectResults) queryService
+          .newQuery("select * from /portfolio p where p.ID * '1' = 2").execute();
+      fail();
+    } catch (QueryInvocationTargetException qite) {
+      assertEquals("Arithmetic Operations can only be applied to numbers", qite.getMessage());
+    }
+  }
+
+  @Test
+  public void multiplyOperationShouldThrowExceptionWhenAddingNull() throws Exception {
+    try {
+      SelectResults results = (SelectResults) queryService
+          .newQuery("select * from /portfolio p where p.ID * null = 2").execute();
+      fail();
+    } catch (QueryInvocationTargetException qite) {
+      assertEquals("Cannot evaluate arithmetic operations on null values", qite.getMessage());
+    }
+  }
+
+  @Test
+  public void multiplyOperationShouldCorrectlyExecuteInQuery() throws Exception {
+    SelectResults results = (SelectResults) queryService
+        .newQuery("select * from /portfolio p where (p.ID * 1) = 3").execute();
+
+    assertEquals(1, results.size());
+  }
+
+  @Test
+  public void divideOperationShouldThrowExceptionWhenAddingNonNumeric() throws Exception {
+    try {
+      SelectResults results = (SelectResults) queryService
+          .newQuery("select * from /portfolio p where p.ID / '1' = 2").execute();
+      fail();
+    } catch (QueryInvocationTargetException qite) {
+      assertEquals("Arithmetic Operations can only be applied to numbers", qite.getMessage());
+    }
+  }
+
+  @Test
+  public void divideOperationShouldThrowExceptionWhenAddingNull() throws Exception {
+    try {
+      SelectResults results = (SelectResults) queryService
+          .newQuery("select * from /portfolio p where p.ID / null = 2").execute();
+      fail();
+    } catch (QueryInvocationTargetException qite) {
+      assertEquals("Cannot evaluate arithmetic operations on null values", qite.getMessage());
+    }
+  }
+
+  @Test
+  public void divideOperationShouldCorrectlyExecuteInQuery() throws Exception {
```

Comment:
tests for division by zero needed ?

Reply:
Sure I can add a test for that

Reply:
What about tests for operator precedence?

`select * from /portfolio p where p.ID + 2 / 3 = 1`

Reply:
Added some operator precedence tests